### DISCUSSION TITLE
Build docker image failed

### DISCUSSION THREAD

**Discussion by jackweiwang** [milvus-io/milvus#16539] [Q&A and General discussion] [answered]
root@sprs-milvus-service-pre:~/milvus# bash build/build_image.sh 
++++ dirname build/build_image.sh
+++ cd build
+++ pwd
++ dirname /root/milvus/build
+ toplevel=/root/milvus
+ OS_NAME=ubuntu18.04
+ MILVUS_IMAGE_REPO=milvusdb/milvus
+ MILVUS_IMAGE_TAG=latest
+ pushd /root/milvus
~/milvus ~/milvus
+ docker build -f ./build/docker/milvus/ubuntu18.04/Dockerfile -t milvusdb/milvus:latest .
Sending build context to Docker daemon  4.362GB
Step 1/14 : FROM milvusdb/openblas:ubuntu18.04-20210428 AS openblas
 ---> 0481017ca014
Step 2/14 : FROM ubuntu:bionic-20200921
 ---> 56def654ec22
Step 3/14 : RUN apt-get update &&     apt-get install -y --no-install-recommends libtbb-dev gfortran netcat iputils-ping ca-certificates &&     apt-get remove --purge -y &&     rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> 76bcb65946af
Step 4/14 : COPY --from=openblas /usr/lib/libopenblas-r0.3.9.so /usr/lib/
 ---> Using cache
 ---> cbb477a46371
Step 5/14 : RUN ln -s /usr/lib/libopenblas-r0.3.9.so /usr/lib/libopenblas.so.0 &&     ln -s /usr/lib/libopenblas.so.0 /usr/lib/libopenblas.so
 ---> Using cache
 ---> 0a10483ba0ed
Step 6/14 : COPY ./bin/ /milvus/bin/
 ---> Using cache
 ---> 71a397774d1f
Step 7/14 : COPY ./configs/ /milvus/configs/
 ---> Using cache
 ---> 45495f6cbe86
Step 8/14 : COPY ./lib/ /milvus/lib/
COPY failed: file not found in build context or excluded by .dockerignore: stat lib/: file does not exist

**Comment by yhmo** [comment, marked as answer]
In this file, `milvus/build/docker/milvus/ubuntu18.04/Dockerfile`, it requires a "lib" folder under the milvus project:
`COPY ./lib/ /milvus/lib/`

But the folder "lib" doesn't exist.
In the `milvus/Makefile`, there is a command to generate "lib" folder:
```
PWD 	  := $(shell pwd)
LIBRARY_PATH := $(PWD)/lib
install: all
	@echo "Installing binary to './bin'"
	@mkdir -p $(GOPATH)/bin && cp -f $(PWD)/bin/milvus $(GOPATH)/bin/milvus
	@mkdir -p $(LIBRARY_PATH) && cp -r -P $(PWD)/internal/core/output/lib/* $(LIBRARY_PATH)
	@echo "Installation successful."
```
We can see this command copy binaries from `milvus/internal/core/output/lib` into `milvus/lib`. But we don't intend to copy the `$(PWD)/bin/milvus` into `$(GOPATH)/bin/milvus`.
So we can:
1. make a folder "lib" under milvus project
2.  copy binaries from `milvus/internal/core/output/lib` into `milvus/lib`
3. run build/build_image.sh again

**Comment by jackweiwang** [comment]
Very detailed answer, thank you very much.
